cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -191,38 +191,57 @@ VAR_LEN_TYPES(IS_NOT_NULL, isnotnull)
 #undef IS_NOT_NULL
 
 FORCE_INLINE
-const char* substr_utf8_int64_int64(int64 context, const char* input, int32 in_len,
-                                    int64 offset64, int64 length, int32* out_len) {
-  if (length <= 0 || input == nullptr || in_len <= 0) {
-    *out_len = 0;
+const char* substr_utf8_int64_int64(int64 context, const char* input, int32 in_data_len,
+                                    int64 offset64, int64 length, int32* out_data_len) {
+  if (length <= 0 || input == nullptr || in_data_len <= 0) {
+    *out_data_len = 0;
     return "";
   }
 
+  int32 num_in_chars;  // num of chars in input string
+  int32* char_start =
+      reinterpret_cast<int32*>(malloc(in_data_len * 4));  // stores start index of chars
```

Comment:
For future reference, you shouldn't use malloc, but gdv_fn_context_arena_malloc

Reply:
right. I need not free the memory if I use gdv_fn_context_arena_malloc, correct?

Reply:
I'm not sure.